catalog/resource_share.go

```diff
@@ -128,6 +128,13 @@ func (si ShareInfo) resourceShareMap() map[string]SharedDataObject {
 	return m
 }
 
+func Equal(a, b SharedDataObject) bool {
```

Comment:
I would either rename it to be more explicit on what we're comparing, or make it a member function of the corresponding struct.

Reply:
Also perhaps rename the parameters? `b` has the meaning of the "after"/goal state?

Reply:
not blocking, but this should be the method on `SharedDataObject` in the follow-up PR